commit c72bac4c563b975aee8be68200d90b561b427e27
Author: Thomas Debesse <dev@illwieckz.net>
Date:   Fri Oct 21 18:18:02 2022 +0200

    patch: luxmark3/luxmark3-oiio-gpstag-redefinition.patch

diff --git a/src/libOpenImageIO/exif.cpp b/src/libOpenImageIO/exif.cpp
index 391f622d..6b1626c0 100644
--- a/src/libOpenImageIO/exif.cpp
+++ b/src/libOpenImageIO/exif.cpp
@@ -209,63 +209,63 @@ static const EXIF_tag_info exif_tag_table[] = {
 
 
 enum GPSTag {
-    GPSTAG_VERSIONID = 0, 
-    GPSTAG_LATITUDEREF = 1,  GPSTAG_LATITUDE = 2,
-    GPSTAG_LONGITUDEREF = 3, GPSTAG_LONGITUDE = 4, 
-    GPSTAG_ALTITUDEREF = 5,  GPSTAG_ALTITUDE = 6,
-    GPSTAG_TIMESTAMP = 7,
-    GPSTAG_SATELLITES = 8,
-    GPSTAG_STATUS = 9,
-    GPSTAG_MEASUREMODE = 10,
-    GPSTAG_DOP = 11,
-    GPSTAG_SPEEDREF = 12, GPSTAG_SPEED = 13,
-    GPSTAG_TRACKREF = 14, GPSTAG_TRACK = 15,
-    GPSTAG_IMGDIRECTIONREF = 16,  GPSTAG_IMGDIRECTION = 17,
-    GPSTAG_MAPDATUM = 18,
-    GPSTAG_DESTLATITUDEREF = 19,  GPSTAG_DESTLATITUDE = 20,
-    GPSTAG_DESTLONGITUDEREF = 21, GPSTAG_DESTLONGITUDE = 22, 
-    GPSTAG_DESTBEARINGREF = 23,   GPSTAG_DESTBEARING = 24,
-    GPSTAG_DESTDISTANCEREF = 25,  GPSTAG_DESTDISTANCE = 26,
-    GPSTAG_PROCESSINGMETHOD = 27,
-    GPSTAG_AREAINFORMATION = 28,
-    GPSTAG_DATESTAMP = 29,
-    GPSTAG_DIFFERENTIAL = 30,
-    GPSTAG_HPOSITIONINGERROR = 31
+    gpstag_VERSIONID = 0, 
+    gpstag_LATITUDEREF = 1,  gpstag_LATITUDE = 2,
+    gpstag_LONGITUDEREF = 3, gpstag_LONGITUDE = 4, 
+    gpstag_ALTITUDEREF = 5,  gpstag_ALTITUDE = 6,
+    gpstag_TIMESTAMP = 7,
+    gpstag_SATELLITES = 8,
+    gpstag_STATUS = 9,
+    gpstag_MEASUREMODE = 10,
+    gpstag_DOP = 11,
+    gpstag_SPEEDREF = 12, gpstag_SPEED = 13,
+    gpstag_TRACKREF = 14, gpstag_TRACK = 15,
+    gpstag_IMGDIRECTIONREF = 16,  gpstag_IMGDIRECTION = 17,
+    gpstag_MAPDATUM = 18,
+    gpstag_DESTLATITUDEREF = 19,  gpstag_DESTLATITUDE = 20,
+    gpstag_DESTLONGITUDEREF = 21, gpstag_DESTLONGITUDE = 22, 
+    gpstag_DESTBEARINGREF = 23,   gpstag_DESTBEARING = 24,
+    gpstag_DESTDISTANCEREF = 25,  gpstag_DESTDISTANCE = 26,
+    gpstag_PROCESSINGMETHOD = 27,
+    gpstag_AREAINFORMATION = 28,
+    gpstag_DATESTAMP = 29,
+    gpstag_DIFFERENTIAL = 30,
+    gpstag_HPOSITIONINGERROR = 31
 };
 
 static const EXIF_tag_info gps_tag_table[] = {
-    { GPSTAG_VERSIONID,		"GPS:VersionID",	TIFF_BYTE, 4 }, 
-    { GPSTAG_LATITUDEREF,	"GPS:LatitudeRef",	TIFF_ASCII, 2 },
-    { GPSTAG_LATITUDE,		"GPS:Latitude",		TIFF_RATIONAL, 3 },
-    { GPSTAG_LONGITUDEREF,	"GPS:LongitudeRef",	TIFF_ASCII, 2 },
-    { GPSTAG_LONGITUDE,		"GPS:Longitude",	TIFF_RATIONAL, 3 }, 
-    { GPSTAG_ALTITUDEREF,	"GPS:AltitudeRef",	TIFF_BYTE, 1 },
-    { GPSTAG_ALTITUDE,		"GPS:Altitude",		TIFF_RATIONAL, 1 },
-    { GPSTAG_TIMESTAMP,		"GPS:TimeStamp",	TIFF_RATIONAL, 3 },
-    { GPSTAG_SATELLITES,	"GPS:Satellites",	TIFF_ASCII, 0 },
-    { GPSTAG_STATUS,		"GPS:Status",		TIFF_ASCII, 2 },
-    { GPSTAG_MEASUREMODE,	"GPS:MeasureMode",	TIFF_ASCII, 2 },
-    { GPSTAG_DOP,		"GPS:DOP",		TIFF_RATIONAL, 1 },
-    { GPSTAG_SPEEDREF,		"GPS:SpeedRef",		TIFF_ASCII, 2 },
-    { GPSTAG_SPEED,		"GPS:Speed",		TIFF_RATIONAL, 1 },
-    { GPSTAG_TRACKREF,		"GPS:TrackRef",		TIFF_ASCII, 2 },
-    { GPSTAG_TRACK,		"GPS:Track",		TIFF_RATIONAL, 1 },
-    { GPSTAG_IMGDIRECTIONREF,	"GPS:ImgDirectionRef",	TIFF_ASCII, 2 },
-    { GPSTAG_IMGDIRECTION,	"GPS:ImgDirection",	TIFF_RATIONAL, 1 },
-    { GPSTAG_MAPDATUM,		"GPS:MapDatum",		TIFF_ASCII, 0 },
-    { GPSTAG_DESTLATITUDEREF,	"GPS:DestLatitudeRef",	TIFF_ASCII, 2 },
-    { GPSTAG_DESTLATITUDE,	"GPS:DestLatitude",	TIFF_RATIONAL, 3 },
-    { GPSTAG_DESTLONGITUDEREF,	"GPS:DestLongitudeRef",	TIFF_ASCII, 2 },
-    { GPSTAG_DESTLONGITUDE,	"GPS:DestLongitude",	TIFF_RATIONAL, 3 }, 
-    { GPSTAG_DESTBEARINGREF,	"GPS:DestBearingRef",	TIFF_ASCII, 2 },
-    { GPSTAG_DESTBEARING,	"GPS:DestBearing",	TIFF_RATIONAL, 1 },
-    { GPSTAG_DESTDISTANCEREF,	"GPS:DestDistanceRef",	TIFF_ASCII, 2 },
-    { GPSTAG_DESTDISTANCE,	"GPS:DestDistance",	TIFF_RATIONAL, 1 },
-    { GPSTAG_PROCESSINGMETHOD,	"GPS:ProcessingMethod",	TIFF_UNDEFINED, 1 },
-    { GPSTAG_AREAINFORMATION,	"GPS:AreaInformation",	TIFF_UNDEFINED, 1 },
-    { GPSTAG_DATESTAMP,		"GPS:DateStamp",	TIFF_ASCII, 0 },
-    { GPSTAG_DIFFERENTIAL,	"GPS:Differential",	TIFF_SHORT, 1 },
-    { GPSTAG_HPOSITIONINGERROR,	"GPS:HPositioningError",TIFF_RATIONAL, 1 },
+    { gpstag_VERSIONID,		"GPS:VersionID",	TIFF_BYTE, 4 }, 
+    { gpstag_LATITUDEREF,	"GPS:LatitudeRef",	TIFF_ASCII, 2 },
+    { gpstag_LATITUDE,		"GPS:Latitude",		TIFF_RATIONAL, 3 },
+    { gpstag_LONGITUDEREF,	"GPS:LongitudeRef",	TIFF_ASCII, 2 },
+    { gpstag_LONGITUDE,		"GPS:Longitude",	TIFF_RATIONAL, 3 }, 
+    { gpstag_ALTITUDEREF,	"GPS:AltitudeRef",	TIFF_BYTE, 1 },
+    { gpstag_ALTITUDE,		"GPS:Altitude",		TIFF_RATIONAL, 1 },
+    { gpstag_TIMESTAMP,		"GPS:TimeStamp",	TIFF_RATIONAL, 3 },
+    { gpstag_SATELLITES,	"GPS:Satellites",	TIFF_ASCII, 0 },
+    { gpstag_STATUS,		"GPS:Status",		TIFF_ASCII, 2 },
+    { gpstag_MEASUREMODE,	"GPS:MeasureMode",	TIFF_ASCII, 2 },
+    { gpstag_DOP,		"GPS:DOP",		TIFF_RATIONAL, 1 },
+    { gpstag_SPEEDREF,		"GPS:SpeedRef",		TIFF_ASCII, 2 },
+    { gpstag_SPEED,		"GPS:Speed",		TIFF_RATIONAL, 1 },
+    { gpstag_TRACKREF,		"GPS:TrackRef",		TIFF_ASCII, 2 },
+    { gpstag_TRACK,		"GPS:Track",		TIFF_RATIONAL, 1 },
+    { gpstag_IMGDIRECTIONREF,	"GPS:ImgDirectionRef",	TIFF_ASCII, 2 },
+    { gpstag_IMGDIRECTION,	"GPS:ImgDirection",	TIFF_RATIONAL, 1 },
+    { gpstag_MAPDATUM,		"GPS:MapDatum",		TIFF_ASCII, 0 },
+    { gpstag_DESTLATITUDEREF,	"GPS:DestLatitudeRef",	TIFF_ASCII, 2 },
+    { gpstag_DESTLATITUDE,	"GPS:DestLatitude",	TIFF_RATIONAL, 3 },
+    { gpstag_DESTLONGITUDEREF,	"GPS:DestLongitudeRef",	TIFF_ASCII, 2 },
+    { gpstag_DESTLONGITUDE,	"GPS:DestLongitude",	TIFF_RATIONAL, 3 }, 
+    { gpstag_DESTBEARINGREF,	"GPS:DestBearingRef",	TIFF_ASCII, 2 },
+    { gpstag_DESTBEARING,	"GPS:DestBearing",	TIFF_RATIONAL, 1 },
+    { gpstag_DESTDISTANCEREF,	"GPS:DestDistanceRef",	TIFF_ASCII, 2 },
+    { gpstag_DESTDISTANCE,	"GPS:DestDistance",	TIFF_RATIONAL, 1 },
+    { gpstag_PROCESSINGMETHOD,	"GPS:ProcessingMethod",	TIFF_UNDEFINED, 1 },
+    { gpstag_AREAINFORMATION,	"GPS:AreaInformation",	TIFF_UNDEFINED, 1 },
+    { gpstag_DATESTAMP,		"GPS:DateStamp",	TIFF_ASCII, 0 },
+    { gpstag_DIFFERENTIAL,	"GPS:Differential",	TIFF_SHORT, 1 },
+    { gpstag_HPOSITIONINGERROR,	"GPS:HPositioningError",TIFF_RATIONAL, 1 },
     { -1, NULL }  // signal end of table
 };
 
@@ -951,7 +951,7 @@ encode_exif (const ImageSpec &spec, std::vector<char> &blob)
         // Add some required Exif tags that wouldn't be in the spec
         static char ver[] = { 2, 2, 0, 0 };
         append_dir_entry (gps_tagmap, gpsdirs, data,
-                          GPSTAG_VERSIONID, TIFF_BYTE, 4, &ver);
+                          gpstag_VERSIONID, TIFF_BYTE, 4, &ver);
         // Sort the gps tag directory
         std::sort (gpsdirs.begin(), gpsdirs.end(), tagcompare());
 
