commit 0109c0c2c67c476f631804b3e5ab646f6e77aea9
Author: Thomas Debesse <dev@illwieckz.net>
Date:   Fri Oct 21 18:21:11 2022 +0200

    patch: luxmark3/luxmark3-luxcore-no-samples.patch

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7fe48a051..9c668e1f3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,15 +128,15 @@ add_subdirectory(src/luxcore)
 #
 ################################################################################
 
-if(NOT APPLE OR OSX_BUILD_DEMOS)
-	add_subdirectory(samples/benchsimple)
-	add_subdirectory(samples/luxcoredemo)
-	add_subdirectory(samples/luxcorescenedemo)
-	add_subdirectory(samples/luxcoreimplserializationdemo)
-endif()
-if(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
-	add_subdirectory(samples/smallluxgpu4)
-endif(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
+# if(NOT APPLE OR OSX_BUILD_DEMOS)
+# 	add_subdirectory(samples/benchsimple)
+# 	add_subdirectory(samples/luxcoredemo)
+# 	add_subdirectory(samples/luxcorescenedemo)
+# 	add_subdirectory(samples/luxcoreimplserializationdemo)
+# endif()
+# if(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
+# 	add_subdirectory(samples/smallluxgpu4)
+# endif(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
 
 
 ################################################################################
