commit bc5df32a335c63f74acbd8f1e5a13c27a3cfedfa
Author: Thomas Debesse <dev@illwieckz.net>
Date:   Fri Oct 21 18:22:16 2022 +0200

    patch: luxmark3/luxmark3-luxmark-catch-context-errors.patch

diff --git a/src/hardwaretree.cpp b/src/hardwaretree.cpp
index 05c9037..7822c36 100644
--- a/src/hardwaretree.cpp
+++ b/src/hardwaretree.cpp
@@ -97,9 +97,18 @@ static void LuxRaysErrorHandler(const char *msg) {
 }
 
 HardwareTreeModel::HardwareTreeModel(MainWindow *w) : QAbstractItemModel() {
+	Context *ctx = nullptr;
+
 	// Retrieve the hardware information with LuxRays
-	Context *ctx = new Context(LuxRaysErrorHandler);
-	const vector<DeviceDescription *> devDescs = ctx->GetAvailableDeviceDescriptions();
+	try {
+		ctx = new Context(LuxRaysErrorHandler);
+	} catch (cl::Error err) {
+		LM_ERROR("OpenCL ERROR: " << err.what() << "(" << err.err() << ")");
+	} catch (runtime_error err) {
+		LM_ERROR("RUNTIME ERROR: " << err.what());
+	} catch (exception err) {
+		LM_ERROR("ERROR: " << err.what());
+	}
 
 	// Build the gui
 	win = w;
@@ -115,6 +124,11 @@ HardwareTreeModel::HardwareTreeModel(MainWindow *w) : QAbstractItemModel() {
 	HardwareTreeItem *oclGPUDev = new HardwareTreeItem("GPUs and Accelerators");
 	oclDev->appendChild(oclGPUDev);
 
+	if (!ctx)
+		return;
+
+	const vector<DeviceDescription *> devDescs = ctx->GetAvailableDeviceDescriptions();
+
 	int index = 0;
 	for (size_t i = 0; i < devDescs.size(); ++i) {
 		DeviceDescription *devDesc = devDescs[i];
