    patch: boost/fix_boost_formatter_overload.patch

diff --git a/libs/locale/src/icu/formatter.cpp b/libs/locale/src/icu/formatter.cpp
--- a/libs/locale/src/icu/formatter.cpp
+++ b/libs/locale/src/icu/formatter.cpp
@@ -58,7 +58,7 @@ puclic:
-            virtual string_type format(int64_t value,size_t &code_points) const
-            {
-                icu::UnicodeString tmp;
-                icu_fmt_->format(value,tmp);
-                code_points=tmp.countChar32();
-                return cvt_.std(tmp);
-            }
+             virtual string_type format(int64_t value,size_t &code_points) const
+             {
+                 icu::UnicodeString tmp;
+                 icu_fmt_->format(::int64_t(value),tmp);
+                 code_points=tmp.countChar32();
+                 return cvt_.std(tmp);
+             }
