
    from https://github.com/illwieckz/flathub/tree/luxmark3

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 42f753061..f165c991c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -144,1 +144,1 @@ add_subdirectory(src/luxcore)
-#if(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
-	add_subdirectory(samples/smallluxgpu4)
-#endif(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
+if(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
+#	add_subdirectory(samples/smallluxgpu4)
+endif(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
